lldb/source/Target/Target.cpp

```diff
@@ -1438,6 +1439,54 @@ static void LoadScriptingResourceForModule(const ModuleSP &module_sp,
                                                   feedback_stream.GetData());
 }
 
+static void LoadTypeSummariesForModule(ModuleSP module_sp) {
+  auto *sections = module_sp->GetSectionList();
+  if (!sections)
+    return;
+
+  llvm::Triple triple = module_sp->GetArchitecture().GetTriple();
+  SectionSP summaries_sp;
+  if (triple.isOSDarwin()) {
+    if (auto data_sp = sections->FindSectionByName(ConstString("__DATA_CONST")))
+      summaries_sp = data_sp->GetChildren().FindSectionByName(
+          ConstString("__lldbsummaries"));
+  } else if (triple.isOSLinux() || triple.isOSWindows()) {
+    summaries_sp = sections->FindSectionByName(ConstString(".lldbsummaries"));
+  }
+
+  if (!summaries_sp)
+    return;
+
+  TypeCategoryImplSP category;
+  DataVisualization::Categories::GetCategory(ConstString("default"), category);
+
+  DataExtractor extractor;
```

Comment:
There should be a comment explaining the file format here?

Reply:
done